setup.py

```diff
@@ -1,13 +1,73 @@
 #!/usr/bin/env python
 # https://github.com/pypa/pip/issues/7953#issuecomment-645133255
+import numpy as np
 import site
 import sys
 
+from pathlib import Path
 from setuptools import setup
 
+# has to happen after the setup import
+from Cython.Build import cythonize  # noqa isort: skip
+
 site.ENABLE_USER_SITE = "--user" in sys.argv[1:]
 
-# Get configuration information from all of the various subpackages.
-# See the docstring for setup_helpers.update_package_files for more
-# details.
-setup(use_scm_version=True)
+
+def get_cython_files():
+    """Find all cython files (*.pyx) in plasmapy."""
+    return list(Path().glob("plasmapy/**/*.pyx"))
+
+
+def build_extensions():
+    """Build a list of "extensions" for `cythonize`-ing`"""
+    # I'm leaving this here as an example of constructing a list of
+    # setuptools.Extension objects that could be passed to cythonize().
+    # This form could allow for custom parameters for each extension.
+    #
+    # extensions = []
+    # for cy_file in get_cython_files():
+    #     mod_name = ".".join(cy_file.with_suffix("").parts)
+    #     extensions.append(Extension(mod_name, [str(cy_file)]))
+
+    extensions = [str(cy_file) for cy_file in get_cython_files()]
+    return extensions
+
+
+if __name__ == "__main__":
+
+    # Get configuration information from all the various subpackages.
+    # See the docstring for setup_helpers.update_package_files for more
+    # details.
+    setup(
+        use_scm_version=True,
+        zip_safe=False,
+        include_dirs=[np.get_include()],
+        ext_modules=cythonize(build_extensions(), language_level=3),
+    )
+
+    # allow for cleaning of --inplace c and shared object files generated
+    # by:
+    #     python setup.py build_ext --inplace
+    #
+    # this get executed with the command
+    #       python setup.py clean
+    #
+    if "clean" in sys.argv:
```

Comment:
`clean` is already a `setup.py` command and I would be wary of reimplementing it. Have you tried playing around with https://stackoverflow.com/questions/30359216/compile-a-cython-project-and-clean and see whether the default `python setup.py clean --all` does the trick?

People are generally moving away from `setup.py` in the python space in general... but that's probably an issue for another day.

Reply:
I'm not overriding `python setup.py clean`, this still runs as expected.  I'm adding an additional setup after the `clean` command is executed.  The `clean` command does not know about, and thus can not remove, `--inplace` shared object files.  All this is doing is seeing if the `clean` command is specified, letting the normal `clean` run, and then our `setup.py` removes the `--inplace` files.

Reply:
Oooooh okay then, gotcha. 👍🏻